core/templates/pages/collection-editor-page/services/collection-editor-state.service.ts

```diff
@@ -18,200 +18,207 @@
-    };
-
-    return {
-      /**
-       * Loads, or reloads, the collection stored by this service given a
-       * specified collection ID. See setCollection() for more information on
-       * additional behavior of this function.
-       */
-      loadCollection: function(collectionId) {
-        _collectionIsLoading = true;
-        EditableCollectionBackendApiService.fetchCollectionAsync(
-          collectionId).then(
-          function(newCollectionObject) {
-            _updateCollection(newCollectionObject);
-            // TODO(#8521): Remove the use of $rootScope.$applyAsync()
-            // once the controller is migrated to angular.
-            $rootScope.$applyAsync();
-          },
-          function(error) {
-            AlertsService.addWarning(
-              error || 'There was an error when loading the collection.');
-            _collectionIsLoading = false;
-          });
-        CollectionRightsBackendApiService.fetchCollectionRightsAsync(
-          collectionId).then(function(newBackendCollectionRightsObject) {
-          _setCollectionRights(newBackendCollectionRightsObject);
-          _collectionIsLoading = false;
-          // TODO(#8521): Remove the use of $rootScope.$applyAsync()
-          // once the controller is migrated to angular.
-          $rootScope.$applyAsync();
-        }, function(error) {
-          AlertsService.addWarning(
-            error ||
-            'There was an error when loading the collection rights.');
-          _collectionIsLoading = false;
-        });
-      },
+  private _updateCollection(newCollectionObject: Collection): void {
+    this._setCollection(newCollectionObject);
+  }
 
-      /**
-       * Returns whether this service is currently attempting to load the
-       * collection maintained by this service.
-       */
-      isLoadingCollection: function() {
-        return _collectionIsLoading;
-      },
+  private _setCollectionRights(collectionRights: CollectionRights): void {
+    this._collectionRights.copyFromCollectionRights(collectionRights);
+  }
 
-      /**
-       * Returns whether a collection has yet been loaded using either
-       * loadCollection() or setCollection().
-       */
-      hasLoadedCollection: function() {
-        return _collectionIsInitialized;
+  /**
+   * Loads, or reloads, the collection stored by this service given a
+   * specified collection ID. See setCollection() for more information on
+   * additional behavior of this function.
+   */
+  loadCollection(collectionId: string, successCallback: () => void): void {
+    this._collectionIsLoading = true;
+    this.editableCollectionBackendApiService.fetchCollectionAsync(
+      collectionId).then(
+      (newCollectionObject) => {
+        this._updateCollection(newCollectionObject);
+        // TODO(#8521): Remove the use of $rootScope.$applyAsync()
+        // once the controller is migrated to angular.
+        // $rootScope.$applyAsync();
+        if (successCallback) {
```

Comment:
This 'if' statement is not required, since we are always passing the successCallback parameter.

Reply:
Done.